File: Demo/HoundifySDK Sample App (Objective-C)/HoundifySDK Sample App (Objective-C)/SettingsViewController.m
//
//  SettingsViewController.m
//  HoundSDK Test Application
//
//  Created by Cyril Austin on 5/22/15.
//  Copyright (c) 2015 SoundHound, Inc. All rights reserved.
//

#import "SettingsViewController.h"
@import HoundifySDK;

#pragma mark - SettingsViewController

@interface SettingsViewController()

@property(nonatomic, strong) IBOutlet UISwitch* enableSpeechSwitch;
@property(nonatomic, strong) IBOutlet UISwitch* enableSpeechActivationDetectionSwitch;
@property(nonatomic, strong) IBOutlet UISwitch* enableEndOfSpeechDetectionSwitch;

@end

@implementation SettingsViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    // Update UI based on current settings
    
    self.enableSpeechSwitch.on = HoundVoiceSearch.instance.enableSpeech;
    self.enableSpeechActivationDetectionSwitch.on = HoundVoiceSearch.instance.enableHotPhraseDetection;
    self.enableEndOfSpeechDetectionSwitch.on = HoundVoiceSearch.instance.enableEndOfSpeechDetection;
}

- (UIStatusBarStyle)preferredStatusBarStyle
{
    return UIStatusBarStyleLightContent;
}

- (IBAction)enableSpeechValueChanged
{
    HoundVoiceSearch.instance.enableSpeech = self.enableSpeechSwitch.on;
}

- (IBAction)enableSpeechActivationDetectionValueChanged
{
    HoundVoiceSearch.instance.enableHotPhraseDetection = self.enableSpeechActivationDetectionSwitch.on;
}

- (IBAction)enableEndOfSpeechDetectionValueChanged
{
    HoundVoiceSearch.instance.enableEndOfSpeechDetection = self.enableEndOfSpeechDetectionSwitch.on;
}

@end

File: Demo/HoundifySDK Sample App (Swift)/HoundifySDK Sample App (Swift)/SettingsViewController.swift
//
//  SettingsViewController.swift
//  HoundifySDK Sample App (Swift)
//
//  Created by Ken Huang on 11/25/16.
//  Copyright © 2016 SoundHound. All rights reserved.
//

import Foundation
import HoundifySDK

class SettingsViewController: UIViewController {
	
	@IBOutlet weak var enableEndOfSpeechDetectionSwitch: UISwitch!
	@IBOutlet weak var enableHotPhraseSwitch: UISwitch!
	@IBOutlet weak var enableSpeechSwitch: UISwitch!
	
	//MARK: ViewController life cycle.
	override func viewWillAppear(_ animated: Bool) {
		super.viewWillAppear(animated)
		
		self.enableSpeechSwitch.isOn = HoundVoiceSearch.instance().enableSpeech
		self.enableHotPhraseSwitch.isOn = HoundVoiceSearch.instance().enableHotPhraseDetection
		self.enableEndOfSpeechDetectionSwitch.isOn = HoundVoiceSearch.instance().enableEndOfSpeechDetection
    }
	
	@IBAction func enableSpeechValueChanged(_ sender: AnyObject) {
		HoundVoiceSearch.instance().enableSpeech = self.enableSpeechSwitch.isOn
	}
	
	@IBAction func enableHotPhraseValueChanged(_ sender: AnyObject) {
		HoundVoiceSearch.instance().enableHotPhraseDetection = self.enableHotPhraseSwitch.isOn
	}
	
	@IBAction func enableEndOfSpeechDetectionValueChanged(_ sender: AnyObject) {
		HoundVoiceSearch.instance().enableEndOfSpeechDetection = self.enableEndOfSpeechDetectionSwitch.isOn
	}
	
	//MARK: Status Bar Style
	override var preferredStatusBarStyle : UIStatusBarStyle {
		return .lightContent
    }
}
